blogposts/inside-rust/01-announcement.md

```diff
@@ -0,0 +1,162 @@
+    unwinding.  For instance, on Microsoft platforms, `longjmp` is implemented
+    as "forced unwinding" (see below)
+
+Currently, when foreign (non-Rust) code invokes Rust code, the behavior of a
+`panic!()` unwind that "escapes" from Rust is explicitly undefined. Similarly,
+when Rust calls a foreign function that unwinds, the behavior once the unwind
+operation encounters Rust frames is undefined. The primary reason for this is
+to ensure that Rust implementations may use their own unwinding mechanism,
+which may not be compatible with the platform-provided "native" unwinding
+mechanism. Currently, however, `rustc` uses the native mechanism, and there are
+no plans to change this.
+
+### Forced unwinding
+
+Platform ABIs can define a special kind of unwinding called "forced unwinding."
+This type of unwinding is never supposed to be intercepted by the language
+runtime; so, for instance, C++ destructors should not be invoked.
+
+There are two common types of forced unwinding:
+
+* On Windows platforms, `longjmp` is implemented as a forced unwind.
+* On UNIX-like platforms, canceling a `phtread` results in a forced unwind.
+
+## Requirements for any cross-language unwinding specification
+
+* Unwinding between Rust functions (and in particular unwinding of Rust panics)
+  may not necessarily use the system unwinding mechanism
+  * In practice, we do use the system mechanism today, but we would like to
+    reserve the freedom to change this
+* If you enable `-Cpanic=abort`, we are able to optimize the size of binaries
+  to remove most code related to unwinding.
+  * In extreme cases, it should be possible to remove **all** traces of
+    unwinding, when you know that it will not occur.
+  * In practice, most "C" functions are never expected to unwind (because they
+    are written in C, for example, and not in C++).
+    * However, because unwinding is now part of most system ABIs, even C
+      functions can unwind -- most notably, `pthread_cancel` can cause all
+      manner of C functions to unwind, including common functions like `read`
+      and `write`.  (For a complete list, search for "cancellation points" in
+      the [pthreads man
+      page](http://man7.org/linux/man-pages/man7/pthreads.7.html).)
+* Changing the behavior from `-Cpanic=unwind` to `-Cpanic=abort` should not
+  cause Undefined Behavior.
+  * However, this may not be tenable, or at least not without making binaries
+    much larger. See the discussion below for more details.
+  * It may, of course, cause programs to abort that used to execute
+    successfully. This could occur if a panic would've been caught and
+    recovered.
+
+## The primary question: should the `"C"` ABI permit unwinding?
+
+The core question that we would like to decide is whether the `"C"` ABI, as
+defined by Rust, should permit unwinding.
+
+This is not a question we expected to be debating. We've long declared that
+unwinding through Rust's `"C" ABI is undefined behavior. In part, this is
```

Comment:
Stray backtick `

Reply:
Ah, that may be why Vim highlighting choked. Thanks for catching it.